tests/clioptions_test.php

```diff
@@ -0,0 +1,98 @@
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Version information
+ *
+ * @package    tool
+ * @subpackage mergeusers
+ * @author     Andrew Hancox <andrewdchancox@googlemail.com>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class clioptions_test extends advanced_testcase {
+    /**
+     * Setup the test.
+     */
+    public function setUp() {
+        global $CFG;
+        require_once("$CFG->dirroot/admin/tool/mergeusers/lib/mergeusertool.php");
+        $this->resetAfterTest(true);
+    }
+
+    /**
+     * Test option to always rollback merges.
+     */
+    public function test_alwaysrollback() {
+        global $DB;
+
+        // Setup two users to merge.
+        $user_remove = $this->getDataGenerator()->create_user();
+        $user_keep = $this->getDataGenerator()->create_user();
+
+        $mut = new MergeUserTool();
+        list($success, $log, $logid) = $mut->merge($user_keep->id, $user_remove->id);
+
+        // Check $user_remove is suspended.
+        $user_remove = $DB->get_record('user', array('id' => $user_remove->id));
+        $this->assertEquals(1, $user_remove->suspended);
+
+        $user_keep = $DB->get_record('user', array('id' => $user_keep->id));
+        $this->assertEquals(0, $user_keep->suspended);
+
+        $user_remove_2 = $this->getDataGenerator()->create_user();
+
+        $config = tool_mergeusers_config::instance();
+        $config->alwaysRollback = true;
+
+        $mut = new MergeUserTool($config);
+
+        $this->setExpectedException('Exception', 'alwaysRollback option is set so rolling back transaction');
+        list($success, $log, $logid) = $mut->merge($user_keep->id, $user_remove_2->id);
+    }
+
+    /**
+     * Test option to always rollback merges.
+     */
+    public function test_debugdb() {
+        global $DB;
+
+        // Setup two users to merge.
+        $user_remove = $this->getDataGenerator()->create_user();
+        $user_keep = $this->getDataGenerator()->create_user();
+
+        $mut = new MergeUserTool();
+        list($success, $log, $logid) = $mut->merge($user_keep->id, $user_remove->id);
+        $this->assertFalse($this->hasOutput());
+
+        // Check $user_remove is suspended.
+        $user_remove = $DB->get_record('user', array('id' => $user_remove->id));
+        $this->assertEquals(1, $user_remove->suspended);
+
+        $user_keep = $DB->get_record('user', array('id' => $user_keep->id));
+        $this->assertEquals(0, $user_keep->suspended);
+
+        $user_remove_2 = $this->getDataGenerator()->create_user();
+
+        $config = tool_mergeusers_config::instance();
+        $config->debugdb = true;
+
+        $mut = new MergeUserTool($config);
+
+        list($success, $log, $logid) = $mut->merge($user_keep->id, $user_remove_2->id);
+
+        $this->expectOutputRegex('/Query took/');
+    }
+}
```

Comment:
for git and comparing source code, it is better to add a last empty line to the document, please.

Reply:
I was testing it with Moodle 3.4 and just it changes the PHPUnit version and I get some errors.

Reply:
Since our last plugin version is for Moodle 3.3, I'l try if it works ok for at last M3.3.